modules/javafx.controls/src/main/java/javafx/scene/control/skin/ButtonBarSkin.java

```diff
@@ -127,6 +130,18 @@ public ButtonBarSkin(final ButtonBar control) {
      *
      **************************************************************************/
 
+    @Override
+    public void dispose() {
+        if (getSkinnable() == null) {
+            return;
+        }
+
+        updateButtonListeners(getSkinnable().getButtons(), false);
```

Comment:
Can we use `ListenerHelper` in `updateButtonListeners` method?
We may need to use `ListernerHelper` of `ButtonSkin`.

Reply:
good question.

I did not want to make drastic structural changes for risk of introducing regression.  Right now, this skin uses the same pattern when dealing with multiple child items - it adds/removes listeners in the constructor, in the dispose() method, and upon any changes to the observable list.  Perhaps not the most optimal solution, but it works and causes no trouble.  I'd rather keep it as is.